src/types/annotator.js

```diff
@@ -13,6 +18,36 @@
  * @prop {DocumentMetadata} document
  */
 
+/**
+ * An object representing the location in a document that an annotation is
+ * associated with.
+ *
+ * @typedef Anchor
+ * @prop {AnnotationData} annotation
+ * @prop {HTMLElement[]} [highlights]
+ * @prop {Range} [range]
+ */
+
+/**
+ * Anchoring implementation for a particular document type (eg. PDF or HTML).
```

Comment:
Is this akin to an interface? Have we "done this" before in our typedef adventures?

Reply:
> Is this akin to an interface? 

Yes, exactly. In TypeScript syntax the same thing would be written using the keyword `interface`:

```ts
interface AnchoringImpl {
  anchor(root: Element, selectors: Selector[], options?: any): Promise<...>;
  describe(root: Element, range: Range): Promise<Selector[]>;
}
```

> Have we "done this" before in our typedef adventures?

Sort-of. Every use of `@typedef` to describe an object is effectively describing an interface because something that requires a particular type can be passed any object that is compatible.